dask_xgboost/core.py

```diff
@@ -98,6 +116,32 @@ def train_part(env, param, list_of_parts, dmatrix_kwargs=None, **kwargs):
     return result
 
 
+def _package_evals(
+    eval_set, sample_weight_eval_set=None, missing=None, n_jobs=None
+):
+    if eval_set is not None:
+        if sample_weight_eval_set is None:
+            sample_weight_eval_set = [None] * len(eval_set)
+        evals = list(
+            xgb.DMatrix(
+                data,
+                label=label,
+                missing=missing,
+                weight=weight,
+                nthread=n_jobs,
+            )
+            for ((data, label), weight) in zip(
```

Comment:
Can you check: what happens if the user does `eval_set=[(x1, y2), (x2, y2)], sample_weight_eval_set=[weight1]`? Do we sielently drop the `(x2, y2)` eval set here, since the lengths don't match?

Would the equivalent mistake raise in xgboost, or do they silently proceed?

Reply:
I'll check.

Reply:
@TomAugspurger I copied this code from xgboost [here](https://github.com/dmlc/xgboost/blob/master/python-package/xgboost/sklearn.py#L376) and then cleaned it up a bit.  The original code will silently proceed.  This code will error.  I'm going to align it with the original code.

Reply:
Actually, no it doesn't error.  They behave the same.  I'll push up another unit test to confirm.

Reply:
Heh, OK.

FWIW, I think raising is the right thing to do... Maybe open an issue with XGBoost to see if they agree? Then you won't need to change anything here.

Reply:
That's what I was wondering.  What's the right thing to do?  Three options, maybe?

1. Silently drop
1. Silently fill right
1. Raise error

Reply:
@TomAugspurger I opened an issue.  Good catch!